File: tests/Web.Tests.Playwright/tests/pages/about.page.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './base.page';

/**
 * About Page Object Model
 */
export class AboutPage extends BasePage {
  readonly pageHeading: Locator;
  readonly pageContent: Locator;

  constructor(page: Page) {
    super(page);
    this.pageHeading = page.locator('h1, h2').first();
    this.pageContent = page.locator('main, article, .content');
  }

  /**
   * Navigate to about page
   */
  async goto(): Promise<void> {
    await super.goto('/about');
  }

  /**
   * Get the page heading text
   */
  async getHeadingText(): Promise<string> {
    return await this.pageHeading.textContent() || '';
  }

  /**
   * Check if page content is visible
   */
  async hasContent(): Promise<boolean> {
    return await this.pageContent.isVisible();
  }
}

File: e2e/Web.Tests.Playwright/PageObjects/AboutPage.cs
namespace Web.Tests.Playwright.PageObjects;

/// <summary>
/// About Page Object Model
/// </summary>
[ExcludeFromCodeCoverage]
public class AboutPage : BasePage
{
	private readonly ILocator _pageHeading;
	private readonly ILocator _pageContent;

	public AboutPage(IPage page) : base(page)
	{
		_pageHeading = page.Locator("h1, h2").First;
		_pageContent = page.Locator("main, article, .content");
	}

	/// <summary>
	/// Navigate to about page
	/// </summary>
	public override async Task GotoAsync(string path = "/")
	{
		await base.GotoAsync("/about");
	}

	/// <summary>
	/// Get the page heading text
	/// </summary>
	public async Task<string> GetHeadingTextAsync()
	{
		return await _pageHeading.TextContentAsync() ?? string.Empty;
	}

	/// <summary>
	/// Check if page content is visible
	/// </summary>
	public async Task<bool> HasContentAsync()
	{
		return await _pageContent.IsVisibleAsync();
	}
}